sql/src/main/java/io/crate/planner/TableStats.java

```diff
@@ -43,6 +46,57 @@ public void updateTableStats(ObjectLongMap<TableIdent> tableStats) {
      * Returns -1 if the table isn't in the cache
      */
     public long numDocs(TableIdent tableIdent) {
-        return tableStats.getOrDefault(tableIdent, -1);
+        return tableStats.getOrDefault(tableIdent, EMPTY_STATS).numDocs;
+    }
+
+    /**
+     * Returns the size of the table in bytes.
+     * <p>
+     * <p>
+     * The returned number isn't an accurate real-time value but a cached value that is periodically updated
+     * </p>
+     * Returns 0 if the table isn't in the cache
+     */
+    private long sizeInBytes(TableIdent tableIdent) {
+        return tableStats.getOrDefault(tableIdent, EMPTY_STATS).sizeInBytes;
+    }
+
+    /**
+     * Returns an estimation (avg) size of each row of the table in bytes.
+     * <p>
+     * <p>
+     * The returned number isn't an accurate real-time value but a cached value that is periodically updated
+     * </p>
+     * Returns -1 if the table isn't in the cache
+     */
+    public int estimatedSizePerRow(TableIdent tableIdent) {
+        long numDocs = numDocs(tableIdent);
+        if (numDocs == -1) {
+            return -1;
+        } else if (numDocs == 0) {
+            return 0;
+        } else {
+            return (int) (sizeInBytes(tableIdent) / numDocs(tableIdent));
```

Comment:
maybe casting to long would be more suitable...

Reply:
I don't think that a single row could be more than max_integer but I can switch it.
Originally I've thought wrong that this number is used as initial capacity argument for some structure...